core/server/src/main/java/alluxio/master/block/BlockMaster.java

```diff
@@ -235,31 +234,29 @@ public void start(boolean isLeader) throws IOException {
    * @return the number of workers
    */
   public int getWorkerCount() {
-    synchronized (mWorkers) {
-      return mWorkers.size();
-    }
+    return mWorkers.size();
   }
 
   /**
    * @return a list of {@link WorkerInfo} objects representing the workers in Alluxio
    */
   public List<WorkerInfo> getWorkerInfoList() {
-    synchronized (mWorkers) {
-      List<WorkerInfo> workerInfoList = new ArrayList<WorkerInfo>(mWorkers.size());
-      for (MasterWorkerInfo masterWorkerInfo : mWorkers) {
+    List<WorkerInfo> workerInfoList = new ArrayList<>(mWorkers.size());
+    for (MasterWorkerInfo masterWorkerInfo : mWorkers) {
```

Comment:
Question: I notice operations of `IndexSet` are thread-safe. But is iterating over `mWorkers` thread-safe?
e.g., is it possible that we get into infinite-loop in some weird case?